code/script.js

```diff
@@ -1,325 +1,539 @@
-          <button class="filled-button small" onclick="guess('${person.name}')">Guess</button>
+          <button class="filled-button small guess-btn" onclick="guess('${person.name}')" >Guess</button>
         </div>
       </div>
     `
   })
 }
 
-// Randomly select a person from the characters array and set as the value of the variable called secret
+// 3. Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)];
 }
 
-// This function to start (and restart) the game
+
+// 4. This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
+  charactersInPlay = CHARACTERS;
+
   // What else should happen when we start the game?
+  winOrLose.style.display = "none"
+  board.style.display = "flex"
+
+
+  // To load the game interface
+  generateBoard();
+
+  // Get secret person
+  setSecret()
+
+  // To reset the time
+  totalSecond =0;
+
+  // To reset the number of guess an display it
+  numberOfGuess = 0;
+  guessMade.textContent = numberOfGuess;
+
+  // To hide the greeting modal 
+  inputWrapper.style.display = "none"
 }
 
-// setting the currentQuestion object when you select something in the dropdown
+
+
+// 5. setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
 
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+   const value = questions.value;
 
-  currentQuestion = {
+   currentQuestion = {
     category: category,
-    // value: value
+    value: value
   }
 }
 
-// This function should be invoked when you click on 'Find Out' button.
-const checkQuestion = () => {
-  const { category, value } = currentQuestion
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
+// 6. This function should be invoked when you click on 'Find Out' button.
+const checkQuestion = () => {
+  selectQuestion()
+  const {category, value } = currentQuestion //User chooses question
+  // Compare the currentQuestion details with the secret person details in a 
+  // Different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
 
-  } else if (category === 'accessories' || category === 'other') {
+  if (category === "hair" || category === "eyes" ) {
```

Comment:
I split up all categories into separate if-statements, because I heard in a team session someone had trouble with their values getting mixed up. Say for instance the user picks brown, you run the risk of filtering out both brown eyes AND brown hair in this if-statement.